src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -99,7 +99,7 @@ public abstract partial class TypeDesc
         public override int GetHashCode()
         {
             // Inherited types are expected to override
-            return RuntimeHelpers.GetHashCode(this);
+            throw new NotImplementedException();
```

Comment:
Would it make sense to mark the method as `abstract override` to make not overriding this a compile time failure?